tests/testthat/helper-monty.R

```diff
@@ -179,3 +179,74 @@ drop_runner <- function(x) {
   x$restart$runner <- NULL
   x
 }
+
+
+ex_augmented <- function(p_group_1 = 0.25) {
+  x <- local({
+    n <- 100
+
+    m1 <- 0
+    m2 <- 3
+    p <- 0.25
+
+    x1 <- rnorm(n, m1)
+    x2 <- rnorm(n, m2)
+    z <- runif(n) < 0.25
+
+    z * x1 + (1 - z) * x2
+  })
+
+  augmented_data_update <- function(pars, rng) {
+    n <- length(x)
+
+    z_new <- as.integer(monty_random_n_uniform(n, 0, 1, rng) < p_group_1)
+    p <- pars[[1]]
+    m1 <- pars[[2]]
+    m2 <- pars[[3]]
+    density_new <- likelihood_fn(p, m1, m2, x, z_new)
+
+    z <- attr(pars, "data")
+    if (is.null(z)) {
+      z <- z_new
+      density <- density_new
+    } else {
+      ## This recalculates the *old* density, per group.  For an
+      ## expensive model that might not be ideal.
+      density <- likelihood_fn(p, m1, m2, x, z)
+      p_accept <- pmin(1, exp(density_new - density))
```

Comment:
proposal is not symmetric so you need to account for this in the acceptance probability, it should be something like
```
     proposal_ratio <- z * log(p_group_1) + (1 - z) * log(1 - p_group_1) -
        z_new * log(p_group_1) - (1 - z_new) * log(1 - p_group_1)
      p_accept <- pmin(1, exp(density_new - density + proposal_ratio))
```